frontend/src/common/queries/cellCards.ts

```diff
@@ -0,0 +1,449 @@
+import { useMemo } from "react";
+import { useQuery, UseQueryResult } from "react-query";
+import { DEFAULT_FETCH_OPTIONS, JSON_BODY_FETCH_OPTIONS } from "./common";
+import { ENTITIES } from "./entities";
+
+// ontology_tree
```

Comment:
The query functions and associated interfaces for each NextJS API endpoint are organized in blocks. The pattern here is very repetitive, so we might be able to shrink the code dramatically by making a generic handler that spits out the `use[ApiEndpoint]` hooks. 

@ashin-czi Do you want to take a crack at this?